src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -0,0 +1,237 @@
+            `Processing block at height ${height}, hash ${blockHash}`,
+        );
+        const block = await this.getBlock(blockHash, 2);
+        const excludedCoinbaseTxns = block.tx.slice(1);
+        for (const txn of excludedCoinbaseTxns) {
+            const parsedTransaction = await this.parseTransaction(
+                txn,
+                block.hash,
+                block.height,
+            );
+            parsedTransactionList.push(parsedTransaction);
+        }
+        return parsedTransactionList;
+    }
+
+    private async parseTransaction(
+        txn: BlockTransaction,
+        blockHash: string,
+        blockHeight: number,
+    ): Promise<Transaction> {
+        const inputs: TransactionInput[] = await Promise.all(
+            txn.vin.map(this.parseTransactionInput, this),
+        );
+        const outputs: TransactionOutput[] = txn.vout.map(
+            this.parseTransactionOutput,
+            this,
+        );
+
+        return {
+            txid: txn.txid,
+            vin: inputs,
+            vout: outputs,
+            blockHeight,
+            blockHash,
+        };
+    }
+
+    private async parseTransactionInput(
+        txnInput: Input,
+    ): Promise<TransactionInput> {
+        let witness = undefined;
+        let scriptSig = '';
+        const prevTransaction = await this.getRawTransaction(
+            txnInput.txid,
+            true,
+        );
+        const vout = txnInput.vout;
+        const prevOutScript = prevTransaction.vout.find((out) => out.n == vout)
+            .scriptPubKey.hex;
+        witness = txnInput.txinwitness;
+        scriptSig = txnInput.scriptSig.hex;
+
+        return {
+            txid: txnInput.txid,
+            vout,
+            scriptSig,
+            witness,
+            prevOutScript,
+        };
+    }
+
+    private parseTransactionOutput(txnOutput: Output): TransactionOutput {
+        return {
+            scriptPubKey: txnOutput.scriptPubKey.hex,
+            value: txnOutput.value,
+        };
+    }
+
+    private async request(body: RPCRequestBody): Promise<any> {
+        const { rpcUser, rpcPass, rpcPort, rpcHost } =
+            this.configService.get<BitcoinCoreConfig>('bitcoinCore');
+        try {
+            const response = await axios.post(
+                `http://${rpcHost}:${rpcPort}/`,
+                {
+                    ...body,
+                    jsonrpc: '1.0',
+                    id: 'silent_payment_indexer',
+                },
+                {
+                    auth: {
+                        username: rpcUser,
+                        password: rpcPass,
+                    },
+                },
+            );
+            return response.data.result;
+        } catch (error) {
+            this.logger.error(
+                `Request to BitcoinCore failed!\nRequest:\n${JSON.stringify(
+                    body,
+                )}\nError:\n${error.message}`,
+            );
+            throw error;
```

Comment:
here we should handle error properly. If it's an `AxiosError` we should log the response.

Reply:
This would be handled in the axios retry task, that is why i did not include it